django/db/models/query.py

```diff
@@ -992,7 +992,7 @@ def update_or_create(self, defaults=None, create_defaults=None, **kwargs):
                 # This is to maintain backward compatibility as these fields
                 # are not updated unless explicitly specified in the
                 # update_fields list.
-                for field in self.model._meta.local_concrete_fields:
+                for field in self.model._meta.concrete_fields:
```

Comment:
Added a note in the patch but there's a nuance that we should capture here.

Only concrete fields of bases with fields included in `update_fields` should be considered here.

Reply:
Thanks for the review, I answered in the ticket but I'm not sure if you saw it:

In our cases, we have an auto_now field on the parent model only, and we need it to be updated every time, even if there is only a single change in a field from a child model.

From a user perspective, I think having some fields but not others updated would be an unexpected behavior. The fields on the parent model are parts of the child one, it is transparent when we try to access or change them on the instance. I'd expect them to be updated in the same way the child ones are.